clang/include/clang-c/Dependencies.h

```diff
@@ -380,6 +380,210 @@ clang_experimental_DependencyScannerWorker_getFileDependencies_v5(
     CXModuleLookupOutputCallback *MLO, unsigned Options,
     CXFileDependenciesList **Out, CXDiagnosticSet *OutDiags);
 
+/**
+ * Output of \c clang_experimental_DependencyScannerWorker_getDependenciesGraph.
+ */
+typedef struct CXOpaqueDependenciesGraph *CXDependenciesGraph;
+
+/**
+ * An individual module dependency that is part of an overall compilation
+ * \c CXDependenciesGraph.
+ */
+typedef struct CXOpaqueDepGraphModule *CXDepGraphModule;
+
+/**
+ * An individual command-line invocation that is part of an overall compilation
+ * \c CXDependenciesGraph.
+ */
+typedef struct CXOpaqueDepGraphTUCommand *CXDepGraphTUCommand;
+
+/**
+ * Settings to use for the
+ * \c clang_experimental_DependencyScannerWorker_getDependenciesGraph action.
+ */
+typedef struct CXOpaqueDependencyScannerWorkerScanSettings
+    *CXDependencyScannerWorkerScanSettings;
+
+/**
+ * Creates a set of settings for
+ * \c clang_experimental_DependencyScannerWorker_getDependenciesGraph action.
+ * Must be disposed with
+ * \c clang_experimental_DependencyScannerWorkerScanSettings_dispose.
+ *
+ * \param argc the number of compiler invocation arguments (including argv[0]).
+ * \param argv the compiler driver invocation arguments (including argv[0]).
+ * \param ModuleName If non-null, the dependencies of the named module are
+ *                   returned. Otherwise, the dependencies of the whole
+ *                   translation unit are returned.
+ * \param WorkingDirectory the directory in which the invocation runs.
+ * \param MLOContext the context that will be passed to \c MLO each time it is
+ *                   called.
+ * \param MLO a callback that is called to determine the paths of output files
+ *            for each module dependency. This may receive the same module on
+ *            different workers. This should be NULL if
+ *            \c clang_experimental_DependencyScannerService_create_v1 was
+ *            called with \c CXDependencyMode_Flat. This callback will be called
+ *            on the same thread that called \c
+ *            clang_experimental_DependencyScannerWorker_getDependenciesGraph.
+ */
+CINDEX_LINKAGE CXDependencyScannerWorkerScanSettings
+clang_experimental_DependencyScannerWorkerScanSettings_create(
```

Comment:
I see you are not copying the memory for these strings when creating the settings object.  I think that deserves at least a prominent mention in the doc comment, since it's unusual.  Normally I would say just copy the strings, but in this case you can't copy `MLOContext` so it's unclear to me if copying the strings is helpful without that.

Reply:
Added a doc-comment.👍